scenes/music_puzzle/bonfire.gd

```diff
@@ -0,0 +1,8 @@
+class_name Bonfire
+extends StaticBody2D
+
+@onready var fire: AnimatedSprite2D = %Fire
+
+func ignite() -> void:
+	fire.animation = &"burning"
+	fire.play()
```

Comment:
The animation can be passed as a parameter to play as well! (unless there's something I'm missing I think this should be equivalent)
```suggestion
	fire.play(&"burning")
```